lib/msf/core/payload/java/payload_options.rb

```diff
@@ -0,0 +1,32 @@
+# -*- coding: binary -*-
+
+require 'msf/core'
+
+module Msf::Payload::Java::PayloadOptions
+
+  def initialize(info = {})
+    super(info)
+    register_advanced_options(
+      [
+        Msf::OptBool.new('JavaMeterpreterDebug', [ false, "Run the payload in debug mode, with logging enabled" ]),
+        Msf::OptInt.new('Spawn', [true, "Number of subprocesses to spawn", 2])
+      ]
+    )
+  end
+
+  #
+  # Generate default configuration that is to be included in the stager.
+  #
+  def stager_config(opts={})
+    ds = opts[:datastore] || datastore
+    spawn = ds["Spawn"] || 2
+    c =  ""
+    if ds["JavaMeterpreterDebug"]
+      spawn = 0
+      c << "StageParameters=NoRedirect\n"
```

Comment:
Funny, we were just talking about porting https://github.com/rapid7/metasploit-payloads/blob/master/java/javapayload/index.html to Markdown!